src/DistrictResources/DistrictResources.cpp

```diff
@@ -1,4 +1,277 @@
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <valarray>
 #include "DistrictResources.h"
+#include "../Vehicle/FireEngine.h"
+#include "../Vehicle/FireLadder.h"
+#include "../Vehicle/Vehicle.h"
+#include "../Event/Event.h"
 
 namespace LockFreeDispatch {
+
+    void DistrictResources::populateDistrictVehicles(const std::vector<Vehicle>& vehicles)
+    {
+        for (auto vehicle : vehicles)
+        {
+            districtVehicles.push_back(vehicle);
+        }
+    }
+
+    void DistrictResources::populateDistrictFireStations(const std::vector<FireStation>& stations)
+    {
+        for (auto station : stations)
+        {
+            districtFireStations.push_back(station);
+        }
+    }
+
+    FireStation DistrictResources::findFireStation(uint32_t fireStationID) {
+        for (auto station : districtFireStations)
+        {
+            if (station.getFireStationID() == fireStationID)
+            {
+                return station;
+            }
+        }
+        return {};
```

Comment:
Now that you point this out, if line 37 (`return {};`) is reached, something's gone wrong as it has attempted to search for a station that does not and should not exist. I've replaced that with an `exit(-1)` instead.

Reply:
And now that you point that out, it might be worth considering [std::find_if](https://cplusplus.com/reference/algorithm/find_if/) here, too. :slightly_smiling_face: